stdlib/public/runtime/HeapObject.cpp

```diff
@@ -70,7 +70,7 @@ static inline bool isValidPointerForNativeRetain(const void *p) {
   // Check the top of the second byte instead, since Android AArch64 reserves
   // the top byte for its own pointer tagging since Android 11.
   return (intptr_t)((uintptr_t)p << 8) > 0;
-#elif defined(__x86_64__) || defined(__arm64__) || defined(__aarch64__) || defined(_M_ARM64) || defined(__s390x__) || (defined(__powerpc64__) && defined(__LITTLE_ENDIAN__))
+#elif defined(__x86_64__) || defined(__arm64__) || defined(__aarch64__) || defined(_M_ARM64) || defined(__s390x__) || defined(__riscv) || (defined(__powerpc64__) && defined(__LITTLE_ENDIAN__))
```

Comment:
Since the arch support is explicitly for `riscv64` rather than using `arch(riscv)`, please ensure that `__xlen` is `64`.